src/test/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsSupplyMissingValuesStepTest.java

```diff
@@ -0,0 +1,131 @@
+        // given
+        DatasetVersion version = createVersionWithFields(createField(MxrdrMetadataField.DATA_COLLECTION,
+                createField(MxrdrMetadataField.DATA_COLLECTION_ORG_X, VALUE),
+                createField(MxrdrMetadataField.DATA_COLLECTION_ORG_Y, VALUE),
+                createField(MxrdrMetadataField.DATA_COLLECTION_DETECTOR_DISTANCE, VALUE),
+                createField(MxrdrMetadataField.DATA_COLLECTION_OSCILLATION_STEP_SIZE, VALUE),
+                createField(MxrdrMetadataField.DATA_COLLECTION_STARTING_ANGLE, VALUE)));
+        TestDatasetVersionService versionService = new TestDatasetVersionService(version);
+        XdsSupplyMissingValuesStep step = new XdsSupplyMissingValuesStep(Collections.emptyMap(), versionService);
+
+        // when
+        step.runInternal(context, workDir);
+
+        // then
+        List<String> lines = Files.readAllLines(workDir.resolve(XdsAdjustResultStep.XDS_INP));
+
+        //  – values supplied and updated
+        assertThat(lines).anyMatch(keyValMatcher("ORGX", VALUE));
+        assertThat(lines).anyMatch(keyValMatcher("STARTING_ANGLE", VALUE));
+        assertThat(lines).anyMatch(keyValMatcher("DETECTOR_DISTANCE", VALUE));
+
+        //  – values supplied, but other present in file already
+        assertThat(lines).noneMatch(keyValMatcher("OSCILLATION_RANGE", VALUE));
+        assertThat(lines).noneMatch(keyValMatcher("ORGY", VALUE));
+
+        //  – value not supplied nor present in file
+        assertThat(lines).anyMatch(keyValMatcher("X-RAY_WAVELENGTH", "XXX"));
+    }
+
+    // -------------------- PRIVATE --------------------
+
+    private Predicate<String> keyValMatcher(String key, String value) {
+        return s -> s.matches(".*" + key + "=\\s*" + value + ".*");
+    }
+
+    private String getFileFromResources(String name) {
+        return getClass().getClassLoader().getResource(TEST_PATH_PREFIX + name).getPath();
+    }
+
+    private DatasetField createField(MxrdrMetadataField type, String value) {
+        DatasetField field = new DatasetField();
+        DatasetFieldType fieldType = new DatasetFieldType();
+        fieldType.setName(type.getValue());
+        field.setDatasetFieldType(fieldType);
+        field.setValue(value);
+        return field;
+    }
+
+    private DatasetField createField(MxrdrMetadataField type, DatasetField... children) {
+        DatasetField field = createField(type, (String) null);
+        field.setDatasetFieldsChildren(Arrays.asList(children));
+        return field;
+    }
+
+    private DatasetVersion createVersionWithFields(DatasetField... fields) {
+        DatasetVersion version = new DatasetVersion();
+        version.setDatasetFields(Arrays.asList(fields));
+        return version;
+    }
+
+    // -------------------- INNER CLASSES --------------------
+
+    @Vetoed
+    static class TestDatasetVersionService extends DatasetVersionServiceBean {
```

Comment:
This is rather abusing the fact that `DatasetVersionServiceBean` is using field injection instead of proper constructor DI and creating an inconsistent object extension that does not follow parent class contract. It seems like a hacky way to avoid Mockito. Instead you can call troublesome dependency in the beginning of the main tested method and pass datasetVersion object as parameter to another method and call this method in the test instead.